templates/named.dnssec_policy.epp

```diff
@@ -0,0 +1,39 @@
+dnssec-policy <%= $name %> {
+  <%- if $dnskey_ttl { -%>
+    dnskey-ttl <%= $dnskey_ttl %>;
+  <%- } -%>
+  <%- unless $keys.empty { -%>
+    keys {
+      <%- $keys.each |$key| { -%>
+        <%= $key['type'] %> <% if $key['directory'] { %><%= $key['directory']%> <% } %>lifetime <%= $key['lifetime'] %> algorithm <%= $key['algorithm'] %><% if $key['size'] { %> <%= $key['size'] %><% } %>;
+      <%- } -%>
+    };
+  <%- } -%>
+  <%- if $max_zone_ttl { -%>
+    max-zone-ttl <%= $max_zone_ttl %>;
+  <%- } -%>
+  <%- if $parent_ds_ttl { -%>
+    parent-ds-ttl <%= $parent_ds_ttl %>;
+  <%- } -%>
+  <%- if $parent_propagation_delay { -%>
+    parent-propagation-delay <%= $parent_propagation_delay %>;
+  <%- } -%>
+  <%- if $publish_safety { -%>
+    publish-safety <%= $publish_safety %>;
+  <%- } -%>
+  <%- if $retire_safety { -%>
+    retire-safety <%= $retire_safety %>;
+  <%- } -%>
+  <%- if $signatures_refresh { -%>
+    signatures-refresh <%= $signatures_refresh %>;
+  <%- } -%>
+  <%- if $signatures_validity { -%>
+    signatures-validity <%= $signatures_validity %>;
+  <%- } -%>
+  <%- if $signatures_validity_dnskey { -%>
+    signatures-validity-dnskey <%= $signatures_validity_dnskey %>;
+  <%- } -%>
+  <%- if $zone_propagation_delay { -%>
+    zone-propagation-delay <%= $zone_propagation_delay %>;
+  <%- } -%>
```

Comment:
I wonder if it would make sense to pass this as a hash to `epp()` so you don't need so many `if` statements. Either way is fine though.

Reply:
No strong feeling here… This quite looks like PHP but I can live with this and doing the "formatting" on the Puppet side seems less appropriate I think.

Reply:
Perhaps. You could use `$context = { 'keys' => ..., 'options' => { .... } }` but I see your point. Either way works for me.

Reply:
I just pushed a commit to confirm I understood what you say… It feels worse AFAIAC :sob: …

Reply:
Let me know what you prefer; I'll either squash or drop the second commit ;-)